File: frontend/src/api/types/member.ts
import type { Message } from "@/api/types/message.ts"
import type { Platform } from "@/api/types/platform.ts"

export interface Member {
	id: number
	account_id: number
	platform: Platform
	platform_member_id: string
	name: string
	avatar: string
	created_at: string
	updated_at: string
	latest_message?: Message
}

File: server/internal/model/member.go
package model

import (
	"github.com/dromara/carbon/v2"
	"gorm.io/gorm"
)

type Member struct {
	ID               uint64         `gorm:"primary_key;auto_increment" json:"id"`
	AccountID        uint64         `gorm:"uniqueIndex:uniq_acc_member" json:"account_id"`
	Platform         string         `gorm:"index" json:"platform"`
	PlatformMemberId string         `gorm:"size:32;uniqueIndex:uniq_acc_member" json:"platform_member_id"`
	Name             string         `gorm:"size:255" json:"name"`
	Avatar           string         `gorm:"size:255" json:"avatar"`
	CreatedAt        *carbon.Carbon `json:"created_at"`
	UpdatedAt        *carbon.Carbon `json:"updated_at"`
	DeletedAt        gorm.DeletedAt `json:"-"`
}
